data-prepper-plugins/otlp-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/otlp/http/OtlpHttpSender.java

```diff
@@ -0,0 +1,221 @@
+        this.maxRetries = config.getMaxRetries();
+    }
+
+    private static OkHttpClient buildOkHttpClient(final long flushTimeoutMs) {
+        final long httpTimeoutMs = Math.min(
+                Math.max(flushTimeoutMs * 2, 3_000),
+                10_000
+        );
+
+        return new OkHttpClient.Builder()
+                .callTimeout(httpTimeoutMs, TimeUnit.MILLISECONDS)
+                .build();
+    }
+
+    /**
+     * Generates exponential backoff delays with jitter.
+     *
+     * @param retries Number of retries.
+     * @return List of delay durations in milliseconds.
+     */
+    private static List<Integer> generateExponentialBackoffDelays(final int retries) {
+        final List<Integer> delays = new ArrayList<>();
+        for (int i = 0; i < retries; i++) {
+            // Exponential backoff: 100ms, 200ms, 400ms, ...
+            delays.add(BASE_RETRY_DELAY_MS * (1 << i));
+        }
+
+        return delays;
+    }
+
+    /**
+     * Sends the provided OTLP Protobuf payload to the OTLP endpoint.
+     * Retries with exponential backoff and jitter on failure.
+     *
+     * @param payload       The OTLP Protobuf-encoded data to be sent.
+     * @throws IOException  when failed to send the payload.
+     */
+    public void send(@Nonnull final byte[] payload) throws IOException {
+        for (int attempt = 0; attempt <= maxRetries; attempt++) {
+            try {
+                final Optional<byte[]> compressedPayload = gzipCompressor.apply(payload);
+                if (compressedPayload.isEmpty()) {
+                    return;
+                }
+
+                final SdkHttpFullRequest signedRequest = signer.signRequest(compressedPayload.get());
+                final Request.Builder requestBuilder = new Request.Builder()
+                        .url(signedRequest.getUri().toString())
+                        .post(RequestBody.create(compressedPayload.get(), PROTOBUF))
+                        .addHeader("Content-Encoding", "gzip");
+
+                signedRequest.headers().forEach((key, values) -> {
+                    for (final String value : values) {
+                        requestBuilder.addHeader(key, value);
+                    }
+                });
+
+                final Request request = requestBuilder.build();
+
+                final long startTime = System.currentTimeMillis();
+                try (final Response response = httpClient.newCall(request).execute()) {
+                    final long duration = System.currentTimeMillis() - startTime;
+                    sinkMetrics.recordHttpLatency(duration);
+
+                    handleResponse(response);
+
+                    sinkMetrics.incrementPayloadSize(payload.length);
+                    sinkMetrics.incrementPayloadGzipSize(compressedPayload.get().length);
+                    return;
+                }
+            } catch (final IOException ioException) {
```

Comment:
Armeria has a utility that we can use for backoffs and retries. Can you make use of that?

`com.linecorp.armeria.client.retry.Backoff` - We use this in other places in the project.